EASY-1403 files visible to KNOWN not searchable for anonymous

jo-pol opened the pull request:
Fixes EASY-1403 files visible to KNOWN not searchable for anonymous

#### When applied it will
* EASY-1403
* fix some inconveniences discovered when deploying the release candidate (progress logging, less stack traces)
* cleanup as in https://github.com/DANS-KNAW/easy-bag-store/pull/48

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

* [x] See previous PR's and `src/test/resources/README.md`

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

jo-pol committed:
log progress with info per bag, no stack trace if content not indexed

jo-pol committed:
fix file not found for pakbon

jo-pol committed:
dropped obsolete file

jo-pol committed:
don't copy bags to VM as easy-bag-store#48, supply command in a README

https://github.com/DANS-KNAW/easy-bag-store/pull/48

rvanheest commented on src/main/ansible/vagrant.yml:
```diff
@@ -55,42 +55,6 @@
       mode: "0775"
     with_items: "{{ bag_stores }}"
 
-  - name: Check whether the bags exist
```
this removal is after the idea of @janvanmansum in https://github.com/DANS-KNAW/easy-bag-store/pull/48, which I fully support. However, would it make sense to have a symlink to the exact `/vagrant` path? What I mean is: the bags here were put in a certain `stores_dir`; would it make sense to make that a symlink there to `/vagrant/src/test/resources/vault/stores`? Similar for example for projects where we copy test data to from the host to the vm into the `vagrant_home` folder. Would it make sense in those cases to put a symlink to the test data on the host in that `vagrant_home` folder?

Curious to what you think, @janvanmansum and @jo-pol.

janvanmansum replied:
Do you mean something like `/home/vagrant/test-bags` -> `/vagrant/src/test/resources/test-bags`? (Supposing that at that location there actually are test bags ;)

I wouldn't object to that. It would make it a bit easier to find them from the command line.

rvanheest replied:
that is precisely what I mean! Maybe @jo-pol can do that for this PR. I'll look into easy-bag-store.

rvanheest replied:
@jo-pol see this commit (https://github.com/DANS-KNAW/easy-bag-store/pull/52/commits/7778e29af78dc124191aa617f9d448441a2a6004) for an example of such a symlink

rvanheest commented on src/test/resources/vault/stores/pdbs/bags/9da0541a-d2c8-432e-8129-979a9830b427/metadata/files.xml:
```diff
@@ -4,7 +4,7 @@
         <dcterms:format>video/mpeg</dcterms:format>
         <dcterms:title>hubble.mpg</dcterms:title>
-        <dcterms:accessRights>RESTRICTED_GROUP</dcterms:accessRights>
+        <accessibleToRights>RESTRICTED_GROUP</accessibleToRights>
```
when you change this, please also add the specific scheme(s). See for examples `easy-split-multi-deposit` test data.

janvanmansum replied:
@rvanheest : do you mean the XML namespace for `files.xsd`?

rvanheest replied:
yeah, that one. Couldn't think of the word. [Here's the link to the lines I meant.](https://github.com/DANS-KNAW/easy-split-multi-deposit/blob/master/src/test/resources/allfields/output/input-ruimtereis01/bag/metadata/files.xml#L2-L3)

rvanheest reviewed (changes requested):
Looks good! One suggestion for discussion and *only* one change request.

janvanmansum commented on src/test/resources/README.md:
```diff
@@ -0,0 +1,25 @@
+Copy the test bags on the local VM 
+----------------------------------
+
+Make sure to remove eventual files ignored by `.gitignore` such as `.DS_Store`
```
@jo-pol the Dutch "eventuele" is in het Engels *niet* "eventual", maar "possible".

http://www.interglot.com/dictionary/nl/en/search?q=eventuele&m=key
http://www.interglot.com/dictionary/en/nl/search?q=eventual&m=key

janvanmansum reviewed (changes requested):
Mostly OK, but I think the `files.xsd` namespace should be added to the test files (either as the default namespace or with a prefix).

jo-pol committed:
eventual -> possible

jo-pol committed:
provide link to test data on local VM

jo-pol committed:
fix unit test

jo-pol committed:
schemes

jo-pol committed:
EASY-1403 files visible to KNOWN not searchable for anonymous

rvanheest commented on src/main/ansible/vagrant.yml:
```diff
@@ -55,6 +55,13 @@
       mode: "0775"
     with_items: "{{ bag_stores }}"
 
+
```
Extra newline not needed?

jo-pol committed:
review comments

jo-pol committed:
using test bags for manual test on VM

jo-pol commented on src/test/resources/README.md:
````diff
@@ -0,0 +1,29 @@
+find /data/bag-stores/ -name '.*'
+```
+
+Drop the hidden files (if any) and run `easy-update-solr4files-index init`
````
Not very happy with this procedure to share bags between unit/integration test and the manual test environment. However, it is going way beyond the major subjects of this issue.

janvanmansum replied:
I agree it is a bit laborious. It would be easier just to call `easy-bag-store add` a couple of times.

jo-pol commented:
Find out you want logging per file or per bag by playing with the log-level for the following lines:

https://github.com/DANS-KNAW-jp/easy-update-solr4files-index/blob/70da35951c84e4c095c731d838b04fdb1f33e6c2/src/main/scala/nl.knaw.dans.easy.solr4files/components/Solr.scala#L51-L53

jo-pol committed:
app.home as in archetype change

https://github.com/DANS-KNAW/easy-module-archetype/pull/31/commits/26b88d3fb27a0e5643fd8e1fb050aa7670f7489c

PaulBoon commented on src/main/scala/nl.knaw.dans.easy.solr4files/components/FileItem.scala:
```diff
@@ -36,7 +36,8 @@ case class FileItem(bag: Bag, ddm: DDM, xml: Node) extends DebugEnhancedLogging
 
   val path: String = xml.attribute("filepath").map(_.text).getOrElse("")
-  val accessibleTo: String = ( xml \ "accessRights").map(_.text).mkString match {
+  val accessibleTo: String = ( xml \ "accessibleToRights").map(_.text).mkString match {
```
Add a trim, just to be save

jo-pol committed:
dropped obsolete app.home in ReadmeSpec

jo-pol committed:
trim requested by review
